tests/integration/cattletest/core/test_authorization.py

```diff
@@ -1,6 +1,23 @@
 from common_fixtures import *  # NOQA
 
 
+def test_client_access(clients):
+    typesLen = {
+        'admin': 89,
+        'agent': 8,
+        'user': 68,
+        'agentRegister': 4,
+        'test': 140,
+        'readAdmin': 89,
+        'token': 2,
+        'superadmin': 140,
+        'service': 89,
+        'project': 68,
+    }
+    for tuple in clients.items():
+        assert typesLen[tuple[0]] == len(tuple[1].schema.types.items())
```

Comment:
TF does occur. Then you have to go and figure out what schema has new resources. The idea here being that if you add a new resource you need to add to this test for the new length. That way we are making sure accounts can only access the types of resources expected. At least as many as are expected.